src/app/file-panel.js

```diff
@@ -106,11 +106,13 @@ function filepanel (appAPI, files) {
     `
   }
 
-  var api = new EventManager()
+  var events = new EventManager()
   var element = template()
-  element.api = api
-  fileExplorer.api.register('focus', function (path) {
-    api.trigger('focus', [path])
+  // TODO please do not add custom javascript objects, which have no
+  // relation to the DOM to DOM nodes
+  element.events = events
```

Comment:
I understand.
The element represent's the instance and so the events object is kind of related to the API of that instance.

The contsructor returns one element, and that was until now the DOM node, but because the constructur will from now on take a selector or container element to which to attach to, it is instead possible to return the api object and not the element object from the constructor (or let's say from the factory function) that created the "file-panel" :-)